### PR TITLE
Add missing docs for filebeat.registry.migrate_file

### PR DESCRIPTION
**Pull request by urso**
The `filebeat.registry.migrate_file` has been introduced with 7.0. I just noticed that the doc is still missing.

### CODE REVIEW

File: filebeat/_meta/common.reference.p2.yml [hunk 1]
```diff
@@ -32,6 +32,14 @@
 # batch of events has been published successfully. The default value is 0s.
 #filebeat.registry.flush: 0s
 
+
+# The new registry format uses a directory to store the filebeat state.
```
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# Starting with Filebeat 7.0, the registry uses a new directory format to store
```

File: filebeat/_meta/common.reference.p2.yml [hunk 2]
```diff
@@ -32,6 +32,14 @@
...
+# Filebeat will automatically migrate a 6.x registry file to the new format on startup.
```
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# Filebeat state. After you upgrade, Filebeat will automatically migrate a 6.x
```

File: filebeat/_meta/common.reference.p2.yml (hunk 1 above)
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# Starting with Filebeat 7.0, the registry uses a new directory format to store
```

File: filebeat/_meta/common.reference.p2.yml (hunk 2 above)
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# Filebeat state. After you upgrade, Filebeat will automatically migrate a 6.x
```

File: filebeat/_meta/common.reference.p2.yml [hunk 3]
```diff
@@ -32,6 +32,14 @@
...
+# The filebeat.registry.migrate_file setting can be used to point filebeat to the
```
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# registry file to use the new directory format. If you changed
```

File: filebeat/_meta/common.reference.p2.yml [hunk 4]
```diff
@@ -32,6 +32,14 @@
...
+# location of the old registry file, in case filebeat.registry.path has been
```
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# filebeat.registry.path while upgrading, set filebeat.registry.migrate_file to
```

File: filebeat/_meta/common.reference.p2.yml [hunk 5]
```diff
@@ -32,6 +32,14 @@
...
+# changed when upgrading filebeat.
```
Comment (dedemorton) [inline filebeat/_meta/common.reference.p2.yml]:
```suggestion
# point to the old registry file.
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 6]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
 down processing. Setting `registry.flush` to a value >0s reduces write operations,
 helping Filebeat process more events.
 
+[float]
+==== `registry.migrate_file`
+
+Prior to Filebeat 7.0 the registry is stored in one file only. On startup
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
Prior to Filebeat 7.0 the registry is stored in a singe file. When you upgrade
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 7]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+Filebeat will automatically migrate the old Filebeat 6.x registry file to the
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
to 7.0, Filebeat will automatically migrate the old Filebeat 6.x registry file
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 8]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
 down processing. Setting `registry.flush` to a value >0s reduces write operations,
 helping Filebeat process more events.
 
+[float]
+==== `registry.migrate_file`
+
+Prior to Filebeat 7.0 the registry is stored in one file only. On startup
+Filebeat will automatically migrate the old Filebeat 6.x registry file to the 
+new directory based registry.
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
to use the new directory format. Filebeat looks for the file in the location
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 9]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
specified by `filebeat.registry.path`. If you changed the path while upgrading,
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 10]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+By default the file location to migrate from is
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
set `filebeat.registry.migrate_file` to point to the old registry file.
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 11]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+assumed to be given by `filebeat.registry.path`, but in case the path has been
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion

```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 12]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+changed when upgrading one can use `filebeat.registry.migrate_file`.
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion

```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 13]
```diff
@@ -73,6 +73,26 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+filebeat.registry.path: ${path.data}/registry
+filebeat.registry.migrate_file: /path/to/old/registry_file
+-------------------------------------------------------------------------------------
+
+Filebeat will use `filebeat.registry.migrate_file` only if the regisry does not
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]: This is a little ambiguous. Are you saying that the file will only be migrated if a registry that uses the directory format does not yet exist?

Comment (urso) [reply filebeat/docs/filebeat-general-options.asciidoc]: Yes. When migrating, the old registry file will be moved into it's new location.

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 14]
```diff
@@ -73,6 +73,24 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
 down processing. Setting `registry.flush` to a value >0s reduces write operations,
 helping Filebeat process more events.
 
+[float]
+==== `registry.migrate_file`
+
+Prior to Filebeat 7.0 the registry is stored in a singe file. When you upgrade
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
Prior to Filebeat 7.0 the registry is stored in a single file. When you upgrade
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 15]
```diff
@@ -73,6 +73,24 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+specified by `filebeat.registry.path`. If you changed the path while upgrading,
+set `filebeat.registry.migrate_file` to point to the old registry file.
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+filebeat.registry.path: ${path.data}/registry
+filebeat.registry.migrate_file: /path/to/old/registry_file
+-------------------------------------------------------------------------------------
+
+The file will be moved to it's new location only if the new registry using the
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
The registry will be migrated to the new location only if a registry using the
```

File: filebeat/docs/filebeat-general-options.asciidoc [hunk 16]
```diff
@@ -73,6 +73,24 @@ NOTE: Filtering out a huge number of logs can cause many registry updates, slowi
...
+directory format does not exist.
```
Comment (dedemorton) [inline filebeat/docs/filebeat-general-options.asciidoc]:
```suggestion
directory format does not already exist.
```

**Pull request merged**